## ANNOTATION OF SNPs & in/dels

# 1 . Lablog to modify VCF ID field before running VEP.
cat ../../../../samples_id.txt | xargs -I @@ echo "awk 'BEGIN{FS=\"\t\";OFS=\"\t\"} {if( \$0 ~ /^#/ ){print \$0}else{printf \"%s\t%s\t%s\t\", \$1,\$2,\$1\"_\"\$2\"_\"\$4\"_\"\$5 ; for (i=4; i<=NF; i++){printf \"%s\t\",\$i} ; printf \"\n\"}}' ../../../03-postprocessing_variants/@@_new.vcf > ./@@_variants_fil_mod.vcf" > _02_vcfmod.sh
cat ../../../../samples_id.txt | xargs -I @@ echo "sed -i 's/\t$//' @@_variants_fil_mod.vcf" >> _02_vcfmod.sh


# 2. Create variant table.
#conda activate nf-core-sarek-2.6-fix-gatk
cat ../../../../samples_id.txt | xargs -I @@ echo "bcftools query -H -f '%CHROM\t%POS\t%ID\t%REF\t%ALT\t%FILTER\t[%GT\t%DP\t%AD\t%GQ\t]\n' ./@@_variants_fil_mod.vcf > @@_variants.table" > _03_variant_table.sh
cat ../../../../samples_id.txt | xargs -I @@ echo "sed -i -r 's/(#|\[[0-9]+\])//g' @@_variants.table;sed -i 's/:/_/g' @@_variants.table;sed -i 's/ //g' @@_variants.table;sed -i 's/\t*$//g' @@_variants.table " >> _03_variant_table.sh


# 3. Lablog for annotating whole genome and exome samples using Variant Effect Predictor (VEP).
#conda activate ensembl-vep

cat ../../../../samples_id.txt | xargs -I % echo "vep --fasta /processing_Data/bioinformatics/references/eukaria/homo_sapiens/hg38/BroadInstitute/genome/Homo_sapiens_assembly38.fasta -i ./%_variants_fil_mod.vcf -o ./%_vep_annot.vcf --cache --offline --dir_cache /processing_Data/bioinformatics/references/eukaria/homo_sapiens/cache_vep/ --everything --assembly GRCh38 --tab --plugin dbNSFP,/processing_Data/bioinformatics/references/eukaria/homo_sapiens/cache_vep/custom_databases/dbNSFP/GRCh38/dbNSFP4.1a_grch38.gz,clinvar_id,clinvar_trait,clinvar_OMIM_id,clinvar_Orphanet_id,HGVSc_snpEff,HGVSp_snpEff,SIFT_score,SIFT_pred,Polyphen2_HDIV_score,Polyphen2_HDIV_pred,Polyphen2_HVAR_score,Polyphen2_HVAR_pred,MutationTaster_score,MutationTaster_pred,MutationAssessor_score,MutationAssessor_pred,FATHMM_score,FATHMM_pred,PROVEAN_score,PROVEAN_pred,VEST4_score,MetaSVM_score,MetaSVM_pred,MetaLR_score,MetaLR_pred,CADD_raw,CADD_phred,CADD_raw_hg19,CADD_phred_hg19,GERP++_NR,GERP++_RS,phyloP100way_vertebrate,phastCons100way_vertebrate" > _04_vep_annotation.sh

cat ../../../../samples_id.txt | xargs -I @@ echo "grep -v '^##' @@_vep_annot.vcf > @@_vep_annot_head.txt" > _05_mod_vcf_vep.sh
cat ../../../../samples_id.txt | xargs -I @@ echo "sed -i 's/#Uploaded_variation/ID/' @@_vep_annot_head.txt" >> _05_mod_vcf_vep.sh

# 4. Merge the output generated by VEP with dbNSFP_ENSG_gene_compl.txt
#conda activate nf-core-sarek-2.6-fix-gatk
cat ../../../../samples_id.txt | xargs -I % echo "Rscript merge_dbNFSP.R %_vep_annot_head.txt dbNSFP_gene_grch38_compl.txt %" >_06_merge_dbnsfp_R.sh

# 5. Once we have merged Vep output with gene notation from dbNSFP,we have to merge again with variants.table file.
#conda activate nf-core-sarek-2.6-fix-gatk
cat ../../../../samples_id.txt | xargs -I % echo 'Rscript merge_parse.R %_variants.table %_vep_dbNSFP.txt %' > _07_merge_vcf_R.sh

